factory/processing/processComponents.go

```diff
@@ -974,6 +974,18 @@ func getGenesisBlockForShard(miniBlocks []*dataBlock.MiniBlock, shardId uint32)
 	return genesisMiniBlocks
 }
 
+func getGenesisIntraShardMiniblocks(miniBlocks []*dataBlock.MiniBlock) []*dataBlock.MiniBlock {
```

Comment:
what about cross-shard miniblocks? Do we save them also?

Reply:
it seems that for history repo, only the intra shard miniblocks are being handled; all miniblocks have been saved into storage separately

Reply:
but do we have all miniblocks stored? Cause otherwise the API route won't be able to return all miniblocks (like the one used for staking transactions (shard X -> meta) )

Reply:
yes, all miniblocks are being stored separately, actually, in history repo we do store only some miniblock hashes

Reply:
ok, great